core/src/main/scala/org/apache/spark/h2o/utils/H2OTypeUtils.scala

```diff
@@ -23,20 +23,25 @@ import water.fvec.Vec
  *
  */
 object H2OTypeUtils {
+  private val ClassMap = Map[Class[_], Byte](
+  classOf[java.lang.Byte] -> Vec.T_NUM,
+  classOf[java.lang.Short] -> Vec.T_NUM,
+  classOf[java.lang.Integer] -> Vec.T_NUM,
+  classOf[java.lang.Long] -> Vec.T_NUM,
+  classOf[java.lang.Float] -> Vec.T_NUM,
+  classOf[java.lang.Double] -> Vec.T_NUM,
+  classOf[java.lang.Boolean] -> Vec.T_NUM,
+  classOf[java.lang.String] -> Vec.T_STR,
+  classOf[java.sql.Timestamp] -> Vec.T_TIME
+  ).withDefault(q => throw new IllegalArgumentException(s"Do not understand type $q"))
+
+  def dataTypeToVecType(t: Class[_]): Byte = ClassMap(t)
+
+  import scala.reflect.runtime.universe._
+  import ReflectionUtils._
+
+  def dataTypeToVecType(t: Type): Byte = {
+    ClassMap(typ(t))
```

Comment:
I like this solution!. It's better to create the map once and then just query it then doing the match all over again.

Thanks !